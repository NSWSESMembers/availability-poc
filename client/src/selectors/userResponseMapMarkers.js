/*
  Give it an array of event response objects and it will return and array of
  react-native-maps/map-marker friendly objects (with some extra data mixed in)
*/

import moment from 'moment';

// eslint-disable-next-line import/prefer-default-export
export const userResponseMapMarkers = (filter, responses) => {
  const mapMarkers = [];
  responses.forEach((r) => {
    if (r.locationLatitude !== null && r.locationLongitude !== null && r.user.id !== filter) {
      mapMarkers.push({
        displayName: r.user.displayName,
        status: r.status,
        destination: r.destination && r.destination.name,
        locationTime: moment.unix(r.locationTime).fromNow(),
        id: r.user.username,
        latitude: r.locationLatitude,
        longitude: r.locationLongitude,
      });
    }
  });
  return mapMarkers;
};
