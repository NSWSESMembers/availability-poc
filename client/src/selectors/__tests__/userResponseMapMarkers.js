import { userResponseMapMarkers } from '../userResponseMapMarkers';

const aUser = [{
  locationLatitude: 10,
  locationLongitude: 10,
  status: 'available',
  destination: { name: 'lhq' },
  locationTime: 1,
  user: {
    id: 1,
    username: 'usera',
    displayName: 'displayName',
  },
}];

const bUser = [{
  locationLatitude: 20,
  locationLongitude: 20,
  destination: { name: 'lhq' },
  locationTime: 1,
  status: 'available',
  user: {
    id: 2,
    username: 'userb',
    displayName: 'displayName',
  },
}];

test('user marker', () => {
  expect(
    userResponseMapMarkers(0, aUser),
  ).toMatchSnapshot();
  expect(
    userResponseMapMarkers(0, aUser.concat(bUser)),
  ).toMatchSnapshot();
});
