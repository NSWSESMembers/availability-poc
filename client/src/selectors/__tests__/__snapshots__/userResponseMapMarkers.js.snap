// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user marker 1`] = `
Array [
  Object {
    "destination": "lhq",
    "displayName": "displayName",
    "id": "usera",
    "latitude": 10,
    "locationTime": "49 years ago",
    "longitude": 10,
    "status": "available",
  },
]
`;

exports[`user marker 2`] = `
Array [
  Object {
    "destination": "lhq",
    "displayName": "displayName",
    "id": "usera",
    "latitude": 10,
    "locationTime": "49 years ago",
    "longitude": 10,
    "status": "available",
  },
  Object {
    "destination": "lhq",
    "displayName": "displayName",
    "id": "userb",
    "latitude": 20,
    "locationTime": "49 years ago",
    "longitude": 20,
    "status": "available",
  },
]
`;
